### ISSUE TITLE
FreeImage 96bpp to 128bpp conversion clamps value

### ISSUE THREAD

**Issue by philcn** [NVIDIAGameWorks/Falcor#209]
When loading a 3-channel exr or hdr file, the FreeImage function that converts the bitmap to 4-channel has a bug where it clamps the color value to 0-1. This is verified by both looking at FreeImage source code, and printing out the bitmap values after loading a 3-channel exr. (The hdr files used by HDRToneMapping sample are such examples).

Relevant code: https://github.com/NVIDIAGameWorks/Falcor/blob/master/Framework/Source/Utils/Bitmap.cpp#L147

**Comment by nbenty** [comment]
We have a fix for that in our 4.0 dev branch.

You can replace the call to `FreeImage_ConvertToRGBAF()` with a call to the following function:

```
    /** Converts 96bpp to 128bpp RGBA without clamping.
        Note that we can't use FreeImage_ConvertToRGBAF() as it clamps to [0,1].
    */
    static FIBITMAP* convertToRGBAF(FIBITMAP* pDib)
    {
        const unsigned width = FreeImage_GetWidth(pDib);
        const unsigned height = FreeImage_GetHeight(pDib);

        auto pNew = FreeImage_AllocateT(FIT_RGBAF, width, height);
        FreeImage_CloneMetadata(pNew, pDib);

        const unsigned src_pitch = FreeImage_GetPitch(pDib);
        const unsigned dst_pitch = FreeImage_GetPitch(pNew);

        const BYTE *src_bits = (BYTE*)FreeImage_GetBits(pDib);
        BYTE* dst_bits = (BYTE*)FreeImage_GetBits(pNew);

        for (unsigned y = 0; y < height; y++)
        {
            const FIRGBF *src_pixel = (FIRGBF*)src_bits;
            FIRGBAF* dst_pixel = (FIRGBAF*)dst_bits;

            for (unsigned x = 0; x < width; x++)
            {
                // Convert pixels directly, while adding a "dummy" alpha of 1.0
                dst_pixel[x].red = src_pixel[x].red;
                dst_pixel[x].green = src_pixel[x].green;
                dst_pixel[x].blue = src_pixel[x].blue;
                dst_pixel[x].alpha = 1.0F;

            }
            src_bits += src_pitch;
            dst_bits += dst_pitch;
        }
        return pNew;
    }
```

**Comment by philcn** [comment]
Great, thanks for the fix.
